### PR TITLE
Update nix install

### PR DESCRIPTION
**Pull request by jocelynthode**
- Update flake.lock
- Update develop environment to latest python to benefit from the cache
- Remove EOL python versions
- Add an overlay output to more easily include the package into top-level pkgs
- rename taxi-zebra plugin from taxi to zebra
- Update taxi-zebra to latest version

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -9,13 +9,29 @@
...
+          {
+            packages = { inherit taxi; };
+            defaultPackage = taxi;
+            devShell = import ./shell.nix { inherit pkgs; };
+          })
+      )
+      {
+        overlay = final: prev: rec {
+          # Prevents collision with taxi from nixpkgs (sftp app)
+          taxi-cli = self.packages.${prev.stdenv.hostPlatform.system}.taxi;
```
Comment (sephii) [inline flake.nix:33]: I think we should avoid pointing base Nix packages to other packages as this could lead to unpleasant surprises (eg. if a Nix package depends on `taxi`).

I’m not sure I understand what problem this change solves: you can already use this Taxi package on your system without any conflict, eg. by importing the Taxi flake and then using it as an overlay on your nixpkgs:

```nix
{
  inputs.taxi = "github:sephii/taxi";
  outputs = { self, nixpkgs, taxi }:
    let
      pkgs = import nixpkgs { overlays = [ (final: prev: { taxi-timesheets = taxi.packages.x86_64-linux.taxi; }) ]; };
    in
    ...
}
```

File: pkgs.nix [hunk 2]
```diff
@@ -4,7 +4,8 @@ let
   lib = pkgs.lib;
   withPlugins = pluginsFunc:
     let plugins = pluginsFunc availablePlugins;
-    in pkgs.python3Packages.buildPythonApplication {
+    in
```
Comment (sephii) [inline pkgs.nix:7]: I’m not sure if there’s a standard in nixpkgs between nixfmt and nixpkgs-fmt, but I’m using nix-mode for Emacs which uses nixfmt, so I would prefer to keep the formatting style of nixfmt. ;)

File: pkgs.nix [hunk 3]
```diff
@@ -55,17 +56,18 @@ let
 
   taxiZebra = pkgs.python3Packages.buildPythonPackage rec {
     pname = "taxi_zebra";
-    version = "2.3.1";
+    version = "3.0.1";
 
-    src = pkgs.python3.pkgs.fetchPypi {
-      inherit pname version;
-      sha256 = "177fzasgchgbixrr4xikfbis8i427qlyb8c93d404rjjny9g7nny";
+    src = pkgs.fetchFromGitHub {
```
Comment (sephii) [inline pkgs.nix:61]: Is there a reason not to fetch the package from PyPI?

File: setup.py [hunk 4]
```diff
@@ -4,39 +4,36 @@
 from taxi import __version__
 
 install_requires = [
-    'click>=3.3',
-    'appdirs>=1.4.0',
+    "click>=3.3",
```
Comment (sephii) [inline setup.py:7]: Please don’t change the formatting in this PR. I would be very happy to blacken the code, but this should be done in a separate PR so we can add the commit to ignore revs.

**Review by sephii — CHANGES_REQUESTED** [review]
Thanks for the PR! It’s nice to see someone took over taxi-zebra. :)

File: setup.py (hunk 4 above)
Comment (jocelynthode) [reply setup.py:7]: Yes sorry, my LSP formats it on its own and sometimes I miss it I'll fix this.

File: pkgs.nix (hunk 3 above)
Comment (jocelynthode) [reply pkgs.nix:61]: This way we can run the tests when building the package.

File: pkgs.nix (hunk 2 above)
Comment (jocelynthode) [reply pkgs.nix:7]: That's my LSP rnix that chooses this mode of formatting I'll change it back.

File: flake.nix (hunk 1 above)
Comment (jocelynthode) [reply flake.nix:33]: The comment was more geared to calling it `taxi-cli` rather than calling it straight `taxi`.

The big advantage of exposing an overlay output to the flake is that it can be consumed by other flakes directly. If you take a look at my current NixOS configuration here: https://github.com/jocelynthode/nixos-config/blob/master/flake.nix#L39 you can see that I use it this way without having to write the overlay myself.

Comment (sephii) [reply flake.nix:33]: Ah yes sorry, I didn’t notice `taxi-cli` refers to `self.packages`.

**Comment by sephii** [comment]
@jocelynthode could you please also update the mentions of Python 3.5 in doc/userguide.rst ?

**Comment by jocelynthode** [comment]
@sephii Done! :)

**Comment by sephii** [comment]
Thanks! :cartwheeling:
